docs/use_cases/retrieval_augmented_generation_eval_qdrant_ragas.md

```diff
@@ -0,0 +1,618 @@
+
+
+### **Synthetic Evaluation Dataset**
+
+Right from the ages of relevance evaluation and assessment of information retrieval systems we have known and acknowledged the importance of the golden set or evaluation dataset.
+Usually , this set contains triplets of queries , document id , relevance scores , in the similar way for RAG evaluation our queries are replaced by questions , document id is replaced by chunk_id or chunk text or an answer or a ground truth and instead of relevance scores - we leverage context texts to provide an optimal evaluation record. Optionally we could add a complexity of the question which is useful for categorising the performance of our RAG system as covered in the first article.
+Addressing the elephant-in-the-room , how do we get on and create one of these evaluation dataset for a problem like RAG.
+
+We got you covered , and here are some techniques you can use to create one of your own.
+
+>  💡 Note: Some of the techniques below may involve using **OpenAI** to generate the questions on the chunked document.
+
+To make some sensible set of questions - we need to use targeted snippets to ensure our questions as well as answers are crisp.
+
+#### 1) **Method 1 - Using T5**
+
+So our first method involves using the [T5](https://huggingface.co/docs/transformers/en/model_doc/t5) model. Make sure to have `pip install transformers` before you use it.
+
+```python
+from transformers import T5ForConditionalGeneration, T5TokenizerFast
+hfmodel = T5ForConditionalGeneration.from_pretrained("ThomasSimonini/t5-end2end-question-generation")
+checkpoint = "t5-base"
+model = T5ForConditionalGeneration.from_pretrained(checkpoint)
+tokenizer = T5TokenizerFast.from_pretrained(checkpoint)
+tokenizer.sep_token = '<sep>'
+tokenizer.add_tokens(['<sep>'])
+model.resize_token_embeddings(len(tokenizer))
+# Check the sep_token_id to verify that it was added to the tokenizer
+tokenizer.sep_token_id
+
+def hf_run_model(input_string, **generator_args):
+  generator_args = {
+  "max_length": 256,
+  "num_beams": 4,
+  "length_penalty": 1.5,
+  "no_repeat_ngram_size": 3,
+  "early_stopping": True,
+  }
+  input_string = "generate questions: " + input_string + " </s>"
+  input_ids = tokenizer.encode(input_string, return_tensors="pt")
+  res = hfmodel.generate(input_ids, **generator_args)
+  output = tokenizer.batch_decode(res, skip_special_tokens=True)
+  output = [item.split("<sep>") for item in output]
+  return output
+```
+
+Using it to create a question for the given text as :
+
+```python
```

Comment:
I suggest making the article a bit less about Qdrant - as seen in the [contributor guide](https://superlinked.com/vectorhub/contributing/contributing).